chapters/mapping_data_to_shaders.md

```diff
@@ -0,0 +1,308 @@
+
+There are various [descriptor types](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VkDescriptorType) in Vulkan, each with their own [detailed description](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#descriptorsets-types) in what they allow.
+
+Descriptors are grouped together in [descriptor sets](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#descriptorsets-sets) which get bound to the shader. Even if there is only a single descriptor in the descriptor set, the entire `VkDescriptorSet` is used when binding to the shader.
+
+### Example
+
+In this example, there are the following 3 descriptor sets:
+
+![mapping_data_to_shaders_descriptor_1.png](../images/mapping_data_to_shaders_descriptor_1.png)
+
+The GLSL of the shader:
+```
+#version 450
+
+layout(set = 0, binding = 0) uniform sampler2D myTextureSampler;
+
+layout(set = 0, binding = 2) uniform uniformBuffer0 {
+    float someData;
+} ubo_0;
+
+layout(set = 0, binding = 3) uniform uniformBuffer1 {
+    float moreData;
+} ubo_1;
+
+layout(set = 2, binding = 0) buffer storageBuffer {
+    float myResults;
+} ssbo;
+```
+
+The corresponding SPIR-V assembly:
+```
+Decorate 19(myTextureSampler) DescriptorSet 0
+Decorate 19(myTextureSampler) Binding 0
+
+MemberDecorate 29(uniformBuffer0) 0 Offset 0
+Decorate 29(uniformBuffer0) Block
+Decorate 31(ubo_0) DescriptorSet 0
+Decorate 31(ubo_0) Binding 2
+
+MemberDecorate 38(uniformBuffer1) 0 Offset 0
+Decorate 38(uniformBuffer1) Block
+Decorate 40(ubo_1) DescriptorSet 0
+Decorate 40(ubo_1) Binding 3
+
+MemberDecorate 44(storageBuffer) 0 Offset 0
+Decorate 44(storageBuffer) BufferBlock
+Decorate 46(ssbo) DescriptorSet 2
+Decorate 46(ssbo) Binding 0
+```
+
+The binding of descriptors is done while recording the command buffer. The descriptors must be bound at the time of a draw/dispatch call. The following is some pseudo code to better represent this:
+```
+vkBeginCommandBuffer();
+// ...
+vkCmdBindPipeline(); // Binds shader
+
+// One possible way of binding the two sets
+vkCmdBindDescriptorSets(firstSet = 0, pDescriptorSets = &descriptor_set_c);
+vkCmdBindDescriptorSets(firstSet = 2, pDescriptorSets = &descriptor_set_b);
+
+vkCmdDraw(); // or dispatch
+// ...
+vkEndCommandBuffer();
+```
+
+The following results would look as followed
+
+![mapping_data_to_shaders_descriptor_2.png](../images/mapping_data_to_shaders_descriptor_2.png)
+
+
+## Push Constants
+
+A push constant is a small bank of values accessible in shaders. Push constants allow the application to set values used in shaders without creating buffers or modifying and binding descriptor sets for each update.
+
```

Comment:
It would be good to explain what you might use them for. Push constants in particular should be recommended for a small amount (a few dwords) of high frequency data (e.g. indices).